### PR TITLE
Improve reconnection behavior on signaling disconnection

### PR DESCRIPTION
**Pull request by hensmi-amazon**
**Issue #:** https://github.com/aws/amazon-chime-sdk-js/issues/2568, other internal requests.

**Description of changes:** 
This consolidates signaling channel reconnection triggering to `MonitorTask`, and adds a new status code `SignalChannelClosedUnexpectedly` to achieve that task. It also fixes reconnection during joins and subscribes, and adds actual bad JOIN message handling instead of the previous, which sorta worked and sorta didn't :).

**Testing:**
Tested reconnection occurs in the following situations:
* On Join (previously would mark this a bad request and not retry)
* On first subscribe (previously would mark this a bad request and not retry)
* on second subscribe
* Mid call, not during subscribe (did not try to reconnect previously)
Also tested that the bad request responses from server in JOIN_ACK were propagated correctly.

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
You can't test these easily besides the mid-call reconnection:
* Join a call
* switch wifi networks
* Connection should immediately reconnect

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
y

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
n

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
n

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- 
+Improve reconnection behavior on signaling disconnection mid call or during join/subscribe
```
Comment (simmkyu) [inline CHANGELOG.md:20]: nit: Move it to the previous line

**Review by simmkyu — COMMENTED** [review]
LGTM overall!

From the JS SDK perspective, there are two types of reconnection ([source](https://github.com/aws/amazon-chime-sdk-js/blob/5497fd178f6837f7dfbe3782cb7c87bbe37fa43f/src/audiovideocontroller/DefaultAudioVideoController.ts#L1309-L1317)):
1. Initial connection to join a meeting.
2. Reconnection during an ongoing session.

Given that the MonitorTask is created at the start, your PR should address both scenarios. But you may include additional test steps specifically for scenario (1).

File: src/audiovideocontroller/DefaultAudioVideoController.ts [hunk 2]
```diff
@@ -1475,21 +1479,21 @@ export default class DefaultAudioVideoController
     }
     if (status.isFailure() || status.isTerminal()) {
       if (this.meetingSessionContext.reconnectController) {
-        const willRetry = this.reconnect(status, error);
-        if (willRetry) {
+        const wiiReconnect = this.reconnect(status, error);
```
[the remark this answers is not in the document]
Comment (hensmi-amazon) [reply src/audiovideocontroller/DefaultAudioVideoController.ts:1482]: typo lol will fix

**Review by hensmi-amazon — COMMENTED** [review]
Will smoke test timed out and refused connections, good point!

**Review by simmkyu — APPROVED** [review]

**Pull request merged**